content/blog/testthat-3-3-0/index.Rmd

```diff
@@ -0,0 +1,172 @@
+
+    ```{r}
+    #| error: true
+
+    test_that("some test", {
+      x <- c(0.408, 0.961, 0.883, 0.46, 0.537, 0.961, 0.851, 0.887, 0.023)
+      expect_all_true(x < 0.95)
+    })
+    ```
+
+*   `expect_disjoint()`, by [@stibu81](https://github.com/stibu81), expects values to be absent:
+
+    ```{r}
+    #| error: true
+
+    test_that("", {
+      expect_disjoint(c("a", "b", "c"), c("c", "d", "e"))
+    })
+    ```
+
+*   `expect_r6_class()` expects an R6 object:
+
+    ```{r}
+    #| error: true
+
+    test_that("", {
+      x <- 10
+      expect_r6_class(x, "foo")
+
+      x <- R6::R6Class("bar")$new()
+      expect_r6_class(x, "foo")
+    })
+    ```
+
+*   `expect_shape()`, by [@michaelchirico](https://github.com/michaelchirico), expects a specific shape (i.e., `nrow()`, `ncol()`, or `dim()`):
+
+    ```{r}
+    #| error: true
+
+    test_that("show off expect_shape() failure messages", {
+      x <- matrix(1:9, nrow = 3)
+      expect_shape(x, nrow = 4)
+      expect_shape(x, dim = c(3, 3, 3))
+      expect_shape(x, dim = c(3, 4))
+    })
+    ```
+
+As you can see from the examples above, when you run a single test interactively (i.e. not as a part of a test suite) you now see exactly how many expectations succeeded and failed.
+
+## Other new features
+
+* testthat generally does a better job of handling nested tests, aka subtests, where you put a `test_that()` inside another `test_that()`, or more typically `it()` inside of `describe()`. Subtests will now generate more informative failure messages, free from duplication, with more informative skips if any subtests don't contain any expectations.
+
+* The snapshot experience has been significantly improved, with all known bugs fixed and some new helpers added: `snapshot_reject()` rejects all modified snapshots by deleting the `.new` variants, and `snapshot_download_gh()` makes it easy to get snapshots off GitHub and into your local package. Additionally, `expect_snapshot()` and friends will now fail when creating a new snapshot on CI, as that's usually a signal that you've forgotten to run the snapshot code locally before committing.
+
+* On CRAN, `test_that()` will automatically skip if a package is not installed, which means that you no longer need to check if suggested packages are installed in your tests.
+
+* `vignette("mocking")` explains mocking in detail, and new `local_mocked_s3_method()`, `local_mocked_s4_method()`, and `local_mocked_r6_class()` make it easier to mock S3 and S4 methods and R6 classes.
+
+* `test_dir()`, `test_check()`, and friends gain a `shuffle` argument that uses `sample()` to randomly reorder the top-level expressions in each test file. This random reordering surfaces dependencies between tests and code outside of any test, as well as dependencies between tests, helping you find and eliminate unintentional dependencies.
+
+* `try_again()` is now publicized, as it's a useful tool for testing flaky code:
```

Comment:
Interesting, might be worth a note that this should still be skipped on CRAN? And include a `skip_on_cran()` in the example.